src/reflection.cpp

```diff
@@ -481,4 +481,232 @@ Offset<const Table *> CopyTable(FlatBufferBuilder &fbb,
   }
 }
 
+bool VerifyStruct(flatbuffers::Verifier &v,
+                  const flatbuffers::Table &parentTable,
+                  voffset_t fieldOffset,
+                  const reflection::Object &obj,
+                  bool isRequired) {
+  assert(obj.is_struct());
+  auto offset = parentTable.GetOptionalFieldOffset(fieldOffset);
+  if (!isRequired) {
+    // Check the actual field.
+    return !offset || v.Verify(reinterpret_cast<const uint8_t*>(&parentTable)
+                               + offset, obj.bytesize());
+  } else {
+    return offset && v.Verify(reinterpret_cast<const uint8_t*>(&parentTable)
+                              + offset, obj.bytesize());
+  }
+}
+
+bool VerifyVectorOfStructs(flatbuffers::Verifier &v,
+                           const flatbuffers::Table &parentTable,
+                           voffset_t fieldOffset,
+                           const reflection::Object &obj,
+                           bool isRequired) {
+  auto p = parentTable.GetPointer<const uint8_t*>(fieldOffset);
+  const uint8_t* end;
+  if (!isRequired) {
+    return !p || v.VerifyVector(p, obj.bytesize(), &end);
+  } else {
+    return p && v.VerifyVector(p, obj.bytesize(), &end);
+  }
+}
+
+// forward declare to resolve cyclic deps between VerifyObject and VerifyVector
+bool VerifyObject(flatbuffers::Verifier &v,
+                  const reflection::Schema &schema,
+                  const reflection::Object &obj,
+                  const flatbuffers::Table *table,
+                  bool isRequired);
+
+bool VerifyVector(flatbuffers::Verifier &v,
+                  const reflection::Schema &schema,
+                  const flatbuffers::Table &table,
+                  const reflection::Field &vecField) {
+  assert(vecField.type()->base_type() == reflection::BaseType::Vector);
+  if (!table.VerifyField<uoffset_t>(v, vecField.offset()))
+    return false;
+
+  switch (vecField.type()->element()) {
+    case reflection::BaseType::None:
+      assert(false);
+      break;
+    case reflection::BaseType::UType:
+      return v.Verify(flatbuffers::GetFieldV<uint8_t>(table, vecField));
```

Comment:
The switch is on the vector element type and this verify function overload is internally calling VerifyVector which is doing exactly what you suggested. Am I missing something?